src/actions.py

```diff
@@ -24,11 +24,17 @@ def perform_operation(_action, target, self):
     if operation in ["center"]:
         args = [self.get_settings("general").get_enum(operation)]
     if operation == "shift":
+        canvas = self.get_window().get_canvas()
+        right_range = (canvas.top_right_axis.get_ylim()[1]
+                       - canvas.top_right_axis.get_ylim()[0])
+        left_range = (canvas.axis.get_ylim()[1]
+                      - canvas.axis.get_ylim()[0])
```

Comment:
I'd rather use figure_settings here, as it is already in scope and should store the exact same values